web/src/lib/components/admin-page/settings/admin-settings.svelte

```diff
@@ -0,0 +1,80 @@
+<svelte:options accessors />
+
+<script lang="ts">
+  import { SystemConfigDto, api } from '@api';
+  import {
+    notificationController,
+    NotificationType,
+  } from '$lib/components/shared-components/notification/notification';
+  import { handleError } from '$lib/utils/handle-error';
+  import _ from 'lodash';
+  import type { SettingsEventType } from './admin-settings';
+  import { createEventDispatcher, onMount } from 'svelte';
+
+  export let config: SystemConfigDto;
+
+  let savedConfig: SystemConfigDto;
+  let defaultConfig: SystemConfigDto;
+
+  const dispatch = createEventDispatcher<{ save: void }>();
+
+  const handleReset = async (detail: SettingsEventType['reset']) => {
+    if (detail.default) {
+      await resetToDefault(detail.configKeys);
+    } else {
+      await reset(detail.configKeys);
+    }
+  };
+
+  const handleSave = async (config: Partial<SystemConfigDto>) => {
+    try {
+      const result = await api.systemConfigApi.updateConfig({
+        systemConfigDto: {
+          ...savedConfig,
+          ...config,
+        },
+      });
+
+      savedConfig = { ...result.data };
+      notificationController.show({ message: 'Settings saved', type: NotificationType.Info });
+
+      dispatch('save');
+    } catch (error) {
+      handleError(error, 'Unable to save settings');
+    }
+  };
+
+  const reset = async (configKeys: Array<keyof SystemConfigDto>) => {
+    const { data: resetConfig } = await api.systemConfigApi.getConfig();
+
+    config = { ...config, ..._.get(resetConfig, configKeys) };
```

Comment:
Why do we need to only reset the selected keys? It seems like reset could reset the entire config object and it would have the same effect.

Reply:
AFAIK if you set `config = resetConfig`, it doesn't work for updating the state

Reply:
Also, if you have changed multiple settings, e.g. storage template and external domain, you don't want to reset storage template when you click on the reset button under external domain. We could of course think about having one global button set, but with the current UI I feel like it's more intuitive to only reset the respective fields

Reply:
Are you saying you make changes in one section and don't save it, then edit a second section?

Reply:
Exactly

Reply:
If they close the section all bets are off and I am fine resetting those values too, especially if the code becomes simpler.

Reply:
Tbh I'd prefer to keep it as is. I don't think it's too complex and it isn't a lot of overhead either